app/models/custom_fields/option_field.rb

```diff
@@ -67,4 +67,15 @@ def option_attributes=(attributes)
       options.where(id: changed[:id]).first.update_attributes(changed)
     }
   end
+
+  def str_to_integer(s)
```

Comment:
It seems that `"string".to_i` returns `0`, and thus

```
s = "string"
(i = s.to_i) >= 0 ? i : nil
```

...returns 0 and not the desired `nil`.